### PR TITLE
Use Space Filling Curve in Threadwise Copy

### PR DESCRIPTION
**Pull request by j4yan**
Added case `SpaceFillingCurve::GetNumAccess()=0` in `GetCoordinateResetStep` didn't deal with the case where `SpaceFillingCurve::GetNumAccess()=0`, which caused the reversion https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/111

This PR should resolve the issue.

### CODE REVIEW

**Comment by qianfengz** [comment]
I found no issue while testing this P.R  with Reduction

**Comment by asroy** [comment]
please fix naming issue

**Comment by asroy** [comment]
please fix naming issue

**Comment by asroy** [comment]
Please fix merge conflict

**Comment by asroy** [comment]
Please fix merge conflict

**Comment by j4yan** [comment]
> 

Done.

**Review by asroy — APPROVED** [review]

File: include/ck/tensor_operation/gpu/thread/threadwise_tensor_slice_transfer.hpp [hunk 1]
```diff
@@ -119,85 +113,26 @@ struct ThreadwiseTensorSliceTransfer_v1r3
...
+        // TODO: Use SpaceFillingCurve::ScalarsPerAccess instread of DstScalarPerVector?
+        static_assert(DstScalarPerVector == SpaceFillingCurve::ScalarPerVector,
+                      "wrong!DstScalarPerVector != SpaceFillingCurve::ScalarPerVector");
+        typename vector_type_maker<DstData, DstScalarPerVector>::type dst_vector;
+        using dst_vector_t = typename vector_type_maker<DstData, DstScalarPerVector>::type::type;
 
-                    static_for<1, i, 1>{}([&](auto j) {
-                        tmp = tmp * ordered_access_lengths[j] + ordered_access_idx[j];
-                    });
+        constexpr auto num_accesses = SpaceFillingCurve::GetNumOfAccess();
```
Comment (asroy) [inline include/ck/tensor_operation/gpu/thread/threadwise_tensor_slice_transfer.hpp:126]: num_access
```Xs``` means a container of ```X```
https://github.com/ROCmSoftwarePlatform/composable_kernel/wiki/Coding-Style

**Review by asroy — APPROVED** [review]

**Pull request merged**